app/uk/gov/hmrc/agentuserclientdetails/services/ClientNameService.scala

```diff
@@ -88,6 +92,40 @@ class ClientNameService @Inject() (
 
   def getPptCustomerName(pptRef: PptRef)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[String]] =
     ifConnector.getPptSubscription(pptRef).map(_.map(_.customerName))
+
+  def createFriendlyNameJobFetchEntry(
```

Comment:
Looks like if maybeArn is None, this method will do nothing.
In this case I think it's better to require the Arn to be supplied (i.e. not an option) and let the caller decide how to handle the case where an Arn is not available.